press/press/doctype/deploy/deploy.py

```diff
@@ -36,9 +41,17 @@ def after_insert(self):
 		self.create_benches()
 
 	def create_benches(self):
-		candidate = frappe.get_cached_doc("Deploy Candidate", self.candidate)
+		deploy_candidate: DeployCandidate = frappe.get_doc("Deploy Candidate", self.candidate)
+		deploy_candidate_builds = frappe.db.get_values(
+			"Deploy Candidate Build", {"deploy_candidate": self.candidate}, pluck="name"
+		)
+
+		# For now let us take the most recent build and use that.
+		build = deploy_candidate_builds[0]
+		deploy_candidate_build: DeployCandidateBuild = frappe.get_doc("Deploy Candidate Build", build)
+
 		environment_variables = [
-			{"key": v.key, "value": v.value} for v in candidate.environment_variables
+			{"key": v.key, "value": v.value} for v in deploy_candidate.environment_variables
 		]
```

Comment:
Better to create two links in `Deploy Candidate` to store one build for x86 and one for arm.

Then while deploying the container, check the architecture of server which one need to be provided.

Reply:
Will do once platform field is added

Reply:
check [this out](https://github.com/frappe/press/pull/2587/commits/19cc8f686d9e893e43e7b4fefdeaf9618c20e655)